### PR TITLE
[openssl] add source vcpkg_from_github

### PR DESCRIPTION
**Pull request by talregev**
**Describe the pull request**

- #### What does your PR fix?
  add source vcpkg_from_github

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  <all / linux, windows, ...>, <Yes/No>

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  `Your answer`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  <Yes / I am still working on this PR>

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/openssl/portfile.cmake [hunk 1]
```diff
@@ -11,13 +11,13 @@ It can be installed on alpine systems via apk add linux-headers.]]
     )
 endif()
 
-set(OPENSSL_VERSION 3.0.3)
+set(OPENSSL_VERSION openssl-3.0.3)
```
Comment (JackBoosY) [inline ports/openssl/portfile.cmake:14]: I noticed that openssl upstream published the new version 3.0.4, can you please also update to that?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/openssl/portfile.cmake (hunk 1 above)
Comment (talregev) [reply ports/openssl/portfile.cmake:14]: No. This is for different PR. It a lot more work.
As you can see here: https://github.com/microsoft/vcpkg/pull/25373 already try  to go next version and it needed to update the port.
So this port is aiming to fix openssl 3.0.3 then we go next version. Thank you.

Comment (talregev) [reply ports/openssl/portfile.cmake:14]: This PR is aiming only to change port source.

**Comment by talregev** [comment]
@JackBoosY Ready for review. Please don't hold this PR, for your PR will need upstream approve.

**Comment by dg0yt** [comment]
> Please don't hold this PR, for your PR will need upstream approve.

But this PR here isn't strictly needed, and still it forces rebuilds on us.

**Comment by talregev** [comment]
> > Please don't hold this PR, for your PR will need upstream approve.
> 
> But this PR here isn't strictly needed, and still it forces rebuilds on us.

I will close this PR once the @JackBoosY openssl 3.0.4 PR will merge.
If it will hold for any reason, My request is to review this PR first.
Thank you.

**Pull request closed without merging**